nototools/gsub_diff.py

```diff
@@ -44,41 +44,144 @@ def __init__(self, file_a, file_b, output_lines=20):
 
     def find_gsub_diffs(self):
         """Report differences in substitution rules."""
-
-        rules_a = self._get_gsub_rules(self.text_a, self.file_a)
-        rules_b = self._get_gsub_rules(self.text_b, self.file_b)
-
-        diffs = []
-        report = ['']  # first line replaced by difference count
-        for rule in rules_a:
-            if rule not in rules_b:
-                diffs.append(('-',) + rule)
-        for rule in rules_b:
-            if rule not in rules_a:
-                diffs.append(('+',) + rule)
-        # ('+', 'smcp', 'Q', 'Q.sc')
+        new = [self._format_rule(r, '+') for r in self.find_new_rules()]
```

Comment:
I see that refactoring those two for loops into new `find_new_rules` and `find_missing_rules` methods make the code more readable, but you are calling the same `_get_gsub_rules` method twice now, which may be an expensive operation. Perhaps you could compute `rules_a` and `rules_b` once and pass them as arguments to the previous methods.. but then again the methods would just become a label for a list comprehension (readable enough from POV). Well, up to you.